### PR TITLE
filter the pods the build controller looks at

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/3169

### CODE REVIEW

**Comment by bparees** [comment]
@deads2k ptal.  note that i left the V(5) message in place, but it should show up almost never now since the controller should (almost?) never see a pod that doesn't have a valid build associated with it.

**Comment by bparees** [comment]
[test]

**Comment by liggitt** [comment]
Shouldn't we wait until the conversation in https://github.com/openshift/origin/issues/2295#issuecomment-115904535 is resolved before adding this?

**Comment by bparees** [comment]
@liggitt my view was "i can fix this now by addin the filtering logic, and if we have to update exactly what we are filtering on later, so be it, but at least the filtering will be in place to be updated"

**Comment by liggitt** [comment]
Ok

**Comment by bparees** [comment]
@soltysh fyi this has bearing on the "remove/alter build pod labels" issue (https://github.com/openshift/origin/issues/2295)

File: pkg/build/controller/factory/factory.go [hunk 1]
```diff
@@ -323,12 +323,16 @@ type podLW struct {
 
 // List lists all Pods.
```
Comment (deads2k) [inline pkg/build/controller/factory/factory.go]: Fix godoc.

**Comment by deads2k** [comment]
update godoc, then lgtm

**Comment by bparees** [comment]
godoc fixed and extraneous commented out code removed.
[merge]

**Pull request merged**